### ISSUE TITLE
`save_hyperparameters` gives a warning incorrectly

### ISSUE THREAD

**Issue by ethanwharris** [PyTorchLightning/pytorch-lightning#12664]
## 🐛 Bug

Save hyperparameters gives a warning about nn.Modules incorrectly when only saving non-module attributes via the `*args`.

### To Reproduce

```python
from torch import nn
from pytorch_lightning import LightningModule


class MyLightningModule(LightningModule):

    def __init__(self, backbone, head, learning_rate):
        super().__init__()

        # Only save the `learning_rate`
        self.save_hyperparameters("learning_rate")

        self.backbone = backbone
        self.head = head
        self.learning_rate = learning_rate


# This line gives two warnings
_ = MyLightningModule(nn.Linear(10, 10), nn.Linear(10, 10), 0.1)
```

### Expected behavior

Shouldn't raise an error as I beleive this example is only saving the learning rate.
